go-controller/pkg/ovn/egressgw.go

```diff
@@ -0,0 +1,254 @@
+package ovn
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog"
+	"net"
+	"strings"
+)
+
+// addPodExternalGW handles detecting if a pod is serving as an external gateway for namespace(s) and adding routes
+// to all pods in that namespace
+func (oc *Controller) addPodExternalGW(pod *kapi.Pod) error {
+	type Network struct {
+		Name string
+		Ips  []string
+	}
+	routingNamespaceAnnotation := pod.Annotations[routingNamespaceAnnotation]
+	if routingNamespaceAnnotation == "" {
+		return nil
+	}
+	klog.Infof("External gateway pod: %s, detected for namespace(s) %s", pod.Name, routingNamespaceAnnotation)
+	var foundGws []net.IP
+	if pod.Spec.HostNetwork {
```

Comment:
This is only returns one IP address in bare metal, Dan Winship has a KEP to fix it.
Mind adding a comment referencing it? https://github.com/kubernetes/enhancements/pull/1665

Reply:
what is the impact here? do we just have a single IP in range pod.Status.PodIPs  ?

Reply:
yeah, I think so, but better if @danwinship can confirm.

Reply:
Yes. I don't think there's any need to comment. The code is correct; it should loop over PodIPs. It's just that in 1.19, PodIPs won't ever have more than 1 element.

Reply:
I was thinking in dual stack about the comment , it won't work with one ip only

Reply:
Yeah but we're not supporting external gateway with dual-stack anyway

Reply:
ahh, then just sorry for the noise :smile:

Reply:
we do need to support dual stack with this. Not right now, but in the future. Is there something I should be using instead of PodIPs?

Reply:
Using PodIPs is correct. It's just that in kube 1.19 hostNetwork pods will only have a single PodIPs value even in a dual-stack cluster, so in OCP 4.6 this would only ever be able to set up the external gateway for a single IP family. But in OCP 4.7, assuming the upstream kube bug has been fixed, then this code will work to set up the external gateway for both families.

If we end up needing to make this work for dual-stack before the upstream hostNetwork PodIPs problem has been fixed then there are things we could do, but it would be ugly and so we shouldn't add that code preemptively if we don't need it yet.